### PR TITLE
`AugAssign` and some `op`s

### PR DESCRIPTION
**Pull request by cyx-6**

### CODE REVIEW

File: python/tvm/script/builder/tir/op.py [hunk 1]
```diff
@@ -107,6 +108,28 @@ def forwarded(*args, **kwargs):
 call_packed = dtype_forward(op.call_packed)
 call_pure_extern = dtype_forward(op.call_pure_extern)
 
+tvm_access_ptr = op_wrapper(op.tvm_access_ptr)
+from tvm.tir.op import tvm_struct_get
```
Comment (junrushao) [inline python/tvm/script/builder/tir/op.py]: move the import logic upwards

File: python/tvm/script/parse/tir/tir.py [hunk 2]
```diff
@@ -80,6 +81,34 @@ def visit_assign(self: Parser, node: doc.Assign) -> None:
         self.eval_assign(target=lhs, source=rhs, bind_value=bind_value)
 
 
+@dispatch.register(token="tir", type_name="AugAssign")
+def visit_aug_ssign(self: Parser, node: doc.AugAssign) -> None:
+    new_rhs = deepcopy(node.target)
+    new_rhs.ctx = doc.Load(
```
Comment (junrushao) [inline python/tvm/script/parse/tir/tir.py]: i guess this is lhs?

Comment (cyx-6) [reply python/tvm/script/parse/tir/tir.py]: yes

File: python/tvm/script/parse/tir/tir.py [hunk 3]
```diff
@@ -80,6 +81,34 @@ def visit_assign(self: Parser, node: doc.Assign) -> None:
...
+            node.value.col_offset,
+            node.value.end_lineno,
+            node.value.end_col_offset,
+        ),
+        node.lineno,
+        node.col_offset,
+        node.lineno,
+        node.end_col_offset,
+    )
+    self.visit_Assign(new_node)
```
Comment (junrushao) [inline python/tvm/script/parse/tir/tir.py]: btw this approach is cool if user inputs are correct; however, it won't produce proper results in error reporting as the column numbers are not accurate.

Instead, would you like to interpret the IR in the following workflow:
1. Evaluate RHS => a
2. Evaluate LHS with `ctx = doc.Load` => b
3. Evaluate `BinOp(npde.op, a, b)` => rhs
4. Evaluate LHS as normal assignment

File: python/tvm/script/parse/tir/tir.py [hunk 4]
```diff
@@ -80,6 +81,48 @@ def visit_assign(self: Parser, node: doc.Assign) -> None:
...
+        node.value.end_lineno,
+        node.value.end_col_offset,
+    )
+    node.target.ctx = doc.Load(*lhs_pos)
+    lhs_name = f"__tvm_tmp_value_aug_assign_lhs_{lhs_pos[0]}"
+    rhs_name = f"__tvm_tmp_value_aug_assign_rhs_{rhs_pos[0]}"
+    lhs = self.eval_expr(node.target)
+    rhs = self.eval_expr(node.value)
+    self.var_table.add(lhs_name, lhs)
+    self.var_table.add(rhs_name, rhs)
```
Comment (junrushao) [inline python/tvm/script/parse/tir/tir.py]: remember to remove those two variables after evaluation...alternatively, shall we have a new var frame?

Comment (cyx-6) [reply python/tvm/script/parse/tir/tir.py]: yes, I have added a temp `with` frame for `var_table`

File: python/tvm/script/parse/tir/tir.py [hunk 5]
```diff
@@ -80,6 +81,49 @@ def visit_assign(self: Parser, node: doc.Assign) -> None:
...
+    rhs_pos = (
+        node.value.lineno,
+        node.value.col_offset,
+        node.value.end_lineno,
+        node.value.end_col_offset,
+    )
+    node.target.ctx = doc.Load(*lhs_pos)
+    with self.var_table.with_frame():
+        lhs_name = f"__tvm_tmp_value_aug_assign_lhs_{lhs_pos[0]}"
+        rhs_name = f"__tvm_tmp_value_aug_assign_rhs_{rhs_pos[0]}"
```
Comment (junrushao) [inline python/tvm/script/parse/tir/tir.py:101]: nit

```suggestion
        lhs_name = f"__tvm_tmp_value_aug_assign_lhs"
        rhs_name = f"__tvm_tmp_value_aug_assign_rhs"
```

**Pull request merged**